libsel4/include/interfaces/sel4.xml

```diff
@@ -235,6 +416,23 @@
                 description="Location of the thread's IPC buffer. Must be 512-byte aligned. The IPC buffer may not cross a page boundary."/>
             <param dir="in" name="bufferFrame" type="seL4_CPtr"
                 description="Capability to a page containing the thread's IPC buffer."/>
+            <error name="seL4_IllegalOperation">
+                <description>
+                    The <texttt text="_service"/> or <texttt text="bufferFrame"/> is a CPtr to a capability of the wrong type.
+                    Or, <texttt text="buffer"/> is not aligned.
+                    Or, <texttt text="bufferFrame"/> is retyped from a device untyped <docref>(see <autoref label="sec:kernmemalloc"/>).</docref>
+                </description>
+            </error>
+            <error name="seL4_InvalidCapability">
+                <description>
+                    The <texttt text="_service"/> is a CPtr to a capability of the wrong type.
+                </description>
+            </error>
+            <error name="seL4_RevokeFirst">
+                <description>
+                    The <texttt text="bufferFrame"/> is a CPtr to a capability of the wrong type.
+                </description>
+            </error>
```

Comment:
This probably the same cause as for `TCBConfigure`. If I've missed it there, please ignore this one as well, but if not, then I don't think `SetIPCBuffer` can generate `seL4_RevokeFirst`

Reply:
Urgh, I finally found it, `deriveCap` for untypeds has an `ensureNoChildren` of course. All good.